packages/host/app/services/store.ts

```diff
@@ -210,11 +214,18 @@ export default class StoreService extends Service implements StoreInterface {
   async add<T extends CardDef>(
     instanceOrDoc: T | LooseSingleCardDocument,
     opts?: {
+      // TODO: apparently this is getting abused by the catalog actions and we
+      // are using this to tell the store the folder _within_ the realm to
+      // upload an instance to, this is not always the actual realm...
       realm?: string;
       relativeTo?: URL | undefined;
       doNotPersist?: true;
     },
   ) {
+    // need to figure out the actual realm because opts.realm is being abused
+    let realmURL = opts?.realm
+      ? this.realm.realmOfURL(new URL(opts.realm))?.href
+      : undefined;
```

Comment:
FYI: we need to clean up the catalog actions so that it is _very_ clear that the thing we are calling the `targetRealm` _is not actually a realm_. Rather this is just a folder in a realm. it's really bad that we are conflating this throughout the context actions , command system, and the store. there will be times in the future were we will want to actually get a handle on the realm but because of how this is named it will be very confusing what is going on. /cc @tintinthong